### ISSUE TITLE
Force user to identify the exact cluster they want

### ISSUE THREAD

**Issue by mmazur** [openshift-online/ocm-cli#138]
*All* cluster commands should refuse to do anything if the provided cluster handle is ambiguous (there's more than one cluster that matches).

This is currently not the case with at least `describe cluster`, possibly more.

**Comment by cben** [comment]
Read-only commands like `describe cluster` could well display all matching clusters.  That would be consistent with `kubectl describe` behavior, and it would be neccessary if we want #108.

OTOH destructive commands like `ocm delete user` should clearly force this :+1:

Hmm, looking at `GetCluster()` helper, it has code to detect multiple matches but alas it requests `Size(1)`, which means it won't see them, just silently pick the first :fearful:  Sending fix...

**Comment by cben** [comment]
Well, it's not as severe as I thought because `name` is unique (per org?), and can't be changed after creation, as are `id` and `external_id`.  (we're not presently matching on display_name.)
So for normal users I don't see a deliberate way to cause a collision, and random collisions extremely unlikely.

But if you are a priviledged internal user seeing multiple orgs, I'm not sure if we enforce `name` uniqueness between orgs?  If not, then this may be bad.  Anyway, fixing.

**Comment by cben** [comment]
Sorry, false alarm :relieved: 
Requesting size=1 returns at most 1 back, but does count them all into the total field:
```
  "page": 1,
  "size": 1,
  "total": 2,
```
and the code of `GetCluster` looks at `total` field :heavy_check_mark: 
I verified by changing the query so it will get multiple matches, and ocm exited with status 1 and message:
```
Error: Failed to get cluster '...': There are 12 clusters with identifier or name '...'
```

**Comment by cben** [comment]
@mmazur are you aware of specific commands, other than `describe`, that behave wrong?

The following command all use `GetCluster()`:
```
cmd/ocm/create/idp/cmd.go
cmd/ocm/create/user/cmd.go
cmd/ocm/create/upgradepolicy/cmd.go
cmd/ocm/tunnel/cmd.go
cmd/ocm/resume/cluster/cmd.go
cmd/ocm/create/machinepool/cmd.go
cmd/ocm/create/ingress/cmd.go
cmd/ocm/hibernate/cluster/cmd.go
cmd/ocm/edit/cluster/cmd.go
cmd/ocm/edit/machinepool/cmd.go
cmd/ocm/list/idp/cmd.go
cmd/ocm/list/addon/cmd.go
cmd/ocm/list/user/cmd.go
cmd/ocm/list/upgradepolicy/cmd.go
cmd/ocm/list/ingress/cmd.go
cmd/ocm/list/machinepool/cmd.go
cmd/ocm/cluster/login/login.go
cmd/ocm/delete/machinepool/cmd.go
cmd/ocm/edit/ingress/cmd.go
cmd/ocm/delete/idp/cmd.go
cmd/ocm/delete/user/cmd.go
cmd/ocm/delete/upgradepolicy/cmd.go
cmd/ocm/delete/ingress/cmd.go
```

**Comment by mmazur** [comment]
Nope, I'm not aware of any more issues. Maybe it's only `cluster describe` that needs a fix.
